### PR TITLE
Add spec for YAML view definition

### PR DESCRIPTION
**Pull request by ramonza**
I am proposing this to be used both in the libraries for now and eventually for OpenCensus service to support runtime view definition.

### CODE REVIEW

**Comment by bogdandrutu** [comment]
The yaml definition looks great, I have a bit of troubles to understand the "runtime support" part. 

Can you please explain me how that will work, I have in mind some use-cases:
1) When a new process start - we need to load a set of views (we can use a yaml file to configure that).
2) Add/Remove one view at runtime - How is this going to be supported in this design? Do I have to get all defined views, then modify what I need and send the whole list of views again?

**Comment by ramonza** [comment]
@bogdandrutu This proposal assumes this will be read from a configuration file. I added a section that says implementations may support reloading the file at runtime.

So yes, if you want to add a view you change the file and then reload it. Concretely, in most setups you will probably store this file somewhere and it will be deployed to the node. For example you could mount it as a configmap in K8s.

File: stats/ViewConfig.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# View Definition
+
+This document describes the YAML configuration for views.
+
+Implementations should provide a way to load this configuration to define views.
+Implementations may provide facilities to:
+* watch the configuration file for changes and reload the views
```
Comment (rakyll) [inline stats/ViewConfig.md]: Is the file here any resource? E.g. metadata from a metadata service?

I'd not call it file to be more precise.

**Comment by ramonza** [comment]
I am going to rationalize this with the view proto. Stay tuned.

**Comment by ramonza** [comment]
Closing this for now, I think it should be handled by making the protobuf suitable for use as config or defining new protobuf that are useful for config.

**Pull request closed without merging**